File: KnowBasicMaths/extract_numbers/extract_numbers.swift
import Foundation

func extractNumbers(_ n: Int) {
    var arr = [Int]()
    var currentN = n
    while currentN > 0 {
        let lastDigit = currentN % 10
        arr.append(lastDigit)
        currentN /= 10
    } 
    print(arr) 
} 

extractNumbers(12345)

File: KnowBasicMaths/extract_numbers/extract_numbers.dart
import 'dart:io';

void extractNumbers(int n) {
  List<int> arr = [];
  int currentN = n;
  while (currentN > 0) {
    int lastDigit = currentN % 10;
    arr.add(lastDigit);
    currentN = currentN ~/ 10; // Use integer division to update n
  }
  print(arr);
}

void main() {
  extractNumbers(12345);
}
